source/articles/2015/save-money-and-be-happier-by-updating-your-gems-every-monday-morning.md

```diff
@@ -0,0 +1,256 @@
+
+author: Adarsh Pandit
+category: writing
+date: 2015-11-12
+logo: ruby_on_rails_logo.png
+published: true
+tags: [ruby on rails, update gems, versions, technical debt, upgrade gems]
+---
+
+As a newcomer to Rails,
+understanding Gems and the Gemfile
+is fairly straightforward:
+You include a gem that you need
+and you get it included everywhere.
+
+We don't often think about it,
+but that last bit is important:
+we are _literally including_
+all of the code from each gem
+into our application, all the time.
+
+There are other implications of
+this `include` behavior which should be considered,
+like speed
+but here I'm going to address one:
+Gem Drift and the associated cost.
+
+## An Example
+
+Let's say I include an auth gem called `huge_bouncer`
+in my app to do authentication and authorization.
+(Note: this is not a real gem, but a name I love.
+Feel free to use it but please attribute here
+so I can feel good about my gem-naming skills.)
+
+Over time, the nice person who wrote `huge_bouncer`
+reviews issues and PRs at night and on weekends
+and keeps development moving at a good pace.
+She is a saint and deserves an award
+or at least you should buy her a beer
+if you run into her at conference.
+
+Newly released gem versions are, for the most part,
+new features and security patches
+(particularly in auth libraries).
+
+Not keeping up to date with new features
+is no big deal in the short term, really.
+I mean, yes, they built a nice OAuth DSL
+but we don't use that in our app, so who cares.
+
+Not keeping up to date with security patches is a bigger issue.
+You know who keeps up with those updates?
+[Bad people who want to target your innocent webapp.][1]
+
+You need to keep your app up to date,
+So how do you do it?
+
+[1]: http://www.phrack.org/papers/attacking_ruby_on_rails.html
+
+## Enter Robots
+Anything you need to do routinely
+should be automated, right?
+You can use [bundler-audit]
+to regularly check for _reported_ security fixes.
+
+[bundler-audit]: https://github.com/rubysec/bundler-audit
+
+It works like this:
+
+Add the gem:
+
+```ruby
+# Gemfile
+# ...
+group :development, :test do
+  gem 'bundler-audit', require: false
+end
+```
+
+Then set it up to check for vulnerabilities
+when you run the test suite:
+
+```ruby
+# Rakefile
+# ...
+task default: [:spec]
+task default: ['bundler:audit']
+```
+
+```ruby
+# lib/tasks/bundler_audit.rake
+if Rails.env.development? || Rails.env.test?
+  require 'bundler/audit/cli'
+
+  namespace :bundler do
+    desc 'Updates the ruby-advisory-db and runs audit'
+    task :audit do
+      %w(update check).each do |command|
+        Bundler::Audit::CLI.start [command]
+      end
+    end
+  end
+end
+```
+
+Be sure to [add it to your CI as well][2].
+
+[2]: /automatically-check-gem-vulnerabilities-in-circle-ci
+
+Note there are [all sorts] of nice
+[paid services] which also do
+this kind of thing.
+
+[all sorts]: https://appcanary.com/
+[paid services]: https://www.datadoghq.com/lpg/
```

Comment:
depbot.com was (is?) another option. It was up as of last week but seems to currently be down.